packages/web-components/src/components/masthead/masthead-composite.ts

```diff
@@ -1326,6 +1327,28 @@ class DDSMastheadComposite extends HostListenerMixin(LitElement) {
       : unauthenticatedCtaButtons;
   }
 
+  /**
+   * A reference to the dds-masthead element.
+   */
+  @query(`${ddsPrefix}-masthead`)
+  mastheadRef;
+
+  /**
+   * Mutation observer
+   */
+  private _heightMutationObserver = new MutationObserver(
+    this._setContainerHeight.bind(this)
+  );
```

Comment:
This throws up a red flag for me after @m4olivei discovered some issues in the carousel component defining an intersectionObserver in a property definition.

I think those issues were limited to that specific use case as we were attempting to set the root to something in the shadowroot which hadn't reliably existed, so I think we're ok, but I'd like Matt's input.

Reply:
This _should_ be OK. If we were actually calling the `this._setContainerHeight` function at this point, that's where we'd be in trouble, b/c then in that function it references `this.mastheadRef`, which since it's a `@query` decorator, would be `null`. As it is, `this._setContainerHeight.bind(this)` just returns a reference to that function bound to `this`. It will actually be called later, at which time it will pull the reference to `this.mastheadRef`, which should be populated by then (it gets populated [after the first `update` lifecycle call](https://codepen.io/m4olivei/pen/JjwLMZx)).

The situation we ran into before ([detailed in PR description here](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/10971#issue-1912120552)), was that a property decorated with `@query` was referenced in a class field initializer, at which point it was `null`.